src/etl/impl/LedgerLoader.hpp

```diff
@@ -197,58 +197,55 @@ class LedgerLoader {
             // asyncWriter consumes from the queue and inserts the data into the
             // Ledger object. Once the below call returns, all data has been pushed
             // into the queue
-            auto [edgeKeys, success] = loadBalancer_->loadInitialLedger(sequence);
-
-            if (success) {
-                size_t numWrites = 0;
-                backend_->cache().setFull();
-
-                auto seconds = ::util::timed<std::chrono::seconds>([this, keys = &edgeKeys, sequence, &numWrites] {
-                    for (auto& key : *keys) {
-                        LOG(log_.debug()) << "Writing edge key = " << ripple::strHex(key);
-                        auto succ = backend_->cache().getSuccessor(*ripple::uint256::fromVoidChecked(key), sequence);
-                        if (succ)
-                            backend_->writeSuccessor(std::move(key), sequence, uint256ToString(succ->key));
-                    }
-
-                    ripple::uint256 prev = data::firstKey;
-                    while (auto cur = backend_->cache().getSuccessor(prev, sequence)) {
-                        ASSERT(cur.has_value(), "Succesor for key {} must exist", ripple::strHex(prev));
-                        if (prev == data::firstKey)
-                            backend_->writeSuccessor(uint256ToString(prev), sequence, uint256ToString(cur->key));
-
-                        if (isBookDir(cur->key, cur->blob)) {
-                            auto base = getBookBase(cur->key);
-                            // make sure the base is not an actual object
-                            if (!backend_->cache().get(base, sequence)) {
-                                auto succ = backend_->cache().getSuccessor(base, sequence);
-                                ASSERT(succ.has_value(), "Book base {} must have a successor", ripple::strHex(base));
-                                if (succ->key == cur->key) {
-                                    LOG(log_.debug()) << "Writing book successor = " << ripple::strHex(base) << " - "
-                                                      << ripple::strHex(cur->key);
-
-                                    backend_->writeSuccessor(
-                                        uint256ToString(base), sequence, uint256ToString(cur->key)
-                                    );
-                                }
+            auto edgeKeys = loadBalancer_->loadInitialLedger(sequence);
```

Comment:
why we don't check 'success' any more? I think it must be checked before we write anything to DB.

Reply:
Because there is infinite loop inside `loadInitialLedger()` until the ledger is successfully loaded.

https://github.com/kuznetsss/clio/blob/5cfc816a178fc25ed13071db9e73882c10e9e005/src/etl/LoadBalancer.cpp#L266-L296